src/components/media/MediaSettingsModal.jsx

```diff
@@ -10,221 +10,184 @@ import { validateFileName } from '../../utils/validators';
+              <img
+                alt={`${media.fileName}`}
+                src={isPendingUpload ? `data:image/png;base64,${content}`
+                  : (
+                    `https://raw.githubusercontent.com/isomerpages/${siteName}/staging/${media.path}${media.path.endsWith('.svg')
+                      ? '?sanitize=true'
+                      : ''}`
+                  )}
               />
             </div>
-            <SaveDeleteButtons
-              saveLabel={isPendingUpload ? "Upload" : "Save"}
-              isDisabled={isPendingUpload ? false : !sha}
-              isSaveDisabled={isPendingUpload ? false : (fileName === this.state.newFileName || errorMessage || !sha)}
-              hasDeleteButton={!isPendingUpload}
-              saveCallback={this.saveFile}
-              deleteCallback={() => this.setState({ canShowDeleteWarningModal: true })}
-              isLoading={isPendingUpload ? false : !sha}
-            />
-          </form>
-        </div>
-        {
-          canShowDeleteWarningModal
-          && (
-            <DeleteWarningModal
-              onCancel={() => this.setState({ canShowDeleteWarningModal: false })}
-              onDelete={this.deleteFile}
-              type="image"
-            />
           )
-        }
+          : (
+            <div className={mediaStyles.editFilePreview}>
+              <p>{media.fileName.split('.').pop().toUpperCase()}</p>
+            </div>
+          )}
+        <form className={elementStyles.modalContent}>
+          <div className={elementStyles.modalFormFields}>
+            <FormField
+              title="File name"
+              value={newFileName}
+              errorMessage={errorMessage}
+              id="file-name"
+              isRequired
+              onFieldChange={(e) => setNewFileName(e.target.value)}
+            />
+          </div>
+          <SaveDeleteButtons
+            saveLabel={isPendingUpload ? "Upload" : "Save"}
+            isDisabled={isPendingUpload ? false : !sha}
+            isSaveDisabled={isPendingUpload ? false : (fileName === newFileName || errorMessage || !sha)}
+            hasDeleteButton={!isPendingUpload}
+            saveCallback={saveHandler}
+            deleteCallback={() => setCanShowDeleteWarningModal(true)}
+            isLoading={isPendingUpload ? false : !sha}
+          />
+        </form>
       </div>
-    );
-  }
+      {
+        canShowDeleteWarningModal
+        && (
+          <DeleteWarningModal
+            onCancel={() => setCanShowDeleteWarningModal(false)}
+            onDelete={deleteHandler}
+            type="image"
+          />
+        )
+      }
+    </div>
+  );
 }
 
+export default MediaSettingsModal
+
 MediaSettingsModal.propTypes = {
   media: PropTypes.shape({
     fileName: PropTypes.string,
     path: PropTypes.string,
     content: PropTypes.string,
   }).isRequired,
-  type: PropTypes.oneOf(['image', 'file']).isRequired,
+  type: PropTypes.oneOf(['images', 'files']).isRequired,
```

Comment:
nit: missing `customPath` as propType

Reply:
Good catch, resolved in https://github.com/isomerpages/isomercms-frontend/pull/453/commits/f07d4c3f03fd5db543ea8fbc1b5911f4573be45e